File: backend/schemas/device.py
from typing import Literal

from pydantic import BaseModel, Field


DeviceStatus = Literal[
    "DISCONNECTED",
    "PAIRING",
    "CONNECTED",
]


class DeviceRequest(BaseModel):
    device_id: str = Field(
        min_length=1,
        max_length=50,
    )


class DeviceConnectionResponse(BaseModel):
    device_id: str
    status: DeviceStatus
    message: str

File: frontend/src/types/device.ts
export type ConnectionStatus =
  | "DISCONNECTED"
  | "CONNECTING"
  | "CONNECTED"
  | "ERROR";

export type DeviceId =
  | "glove_1"
  | "glove_2";

export interface DeviceConnectRequest {
  device_id: DeviceId;
}

export interface DeviceConnectResponse {
  device_id: string;
  status: string;
  message?: string;
}